src/core/src/service_interfaces/StatusHandler.py

```diff
@@ -827,3 +1043,60 @@ def __removed_older_complete_status_files(self, status_folder):
+        """ Calculate final package list size in bytes (because of escape chars) """
+        first_json_dump = json.dumps(package_list)
+
+        return len(json.dumps(first_json_dump).encode("utf-8"))
+
+    def size_of_constant_status_data(self, complete_status_file_payload, assessment_status_index, installation_status_index):
+        """ Get the size in bytes of the complete_status_file without packages data  """
+        status_file_no_list_data = complete_status_file_payload
+        if assessment_status_index is not None:
+            assessment_msg_without_packages = self.__update_substatus_msg(substatus_msg=self.__assessment_substatus_msg_copy, substatus_msg_patches=[])
+            status_file_no_list_data['status']['substatus'][assessment_status_index]['formattedMessage']['message'] = json.dumps(assessment_msg_without_packages)
+
+        if installation_status_index is not None:
+            installation_msg_without_packages = self.__update_substatus_msg(substatus_msg=self.__installation_substatus_msg_copy, substatus_msg_patches=[])
+            status_file_no_list_data['status']['substatus'][installation_status_index]['formattedMessage']['message'] = json.dumps(installation_msg_without_packages)
+
+        return self.__calc_status_size_on_disk(json.dumps(status_file_no_list_data))
+
+    def __get_substatus_index(self, substatus_list_name, substatus_list):
+        """" Get substatus index from the current substatus """
+        for substatus_index, substatus_name in enumerate(substatus_list):
+            if substatus_name['name'] == substatus_list_name:
+                return substatus_index
+
+        return None
+
+    def __recompose_truncated_status_file(self, truncated_status_file, truncated_package_list, truncated_substatus_msg, substatus_index):
+        """ Recompose final truncated status file version """
+        code, _ = self.__get_current_complete_status_errors(substatus_msg=truncated_substatus_msg)
+
+        # Check for existing errors before recompose
+        if code != Constants.PatchOperationTopLevelErrorCode.ERROR:
+            truncated_status_file['status']['substatus'][substatus_index]['status'] = Constants.STATUS_WARNING.lower()      # Update substatus status to warning
+
+        truncated_substatus_msg = self.__update_substatus_msg(substatus_msg=truncated_substatus_msg, substatus_msg_patches=truncated_package_list)
+
+        truncated_status_file['status']['substatus'][substatus_index]['formattedMessage']['message'] = json.dumps(truncated_substatus_msg)
+
+        return truncated_status_file
+
+    def __update_substatus_msg(self, substatus_msg, substatus_msg_patches, substatus_msg_errors=None):
+        substatus_msg['patches'] = substatus_msg_patches
+        if substatus_msg_errors:
+            substatus_msg['errors'] = substatus_msg_errors
+
+        return substatus_msg
+
+    def __get_current_complete_status_errors(self, substatus_msg):
```

Comment:
rename to "__get_errors_from_complete_status_file"

Reply:
On reviewing again, this has nothing to do with complete status file. It is simply fetching error details from substatus_msg that is passed in.

Rename the function name to "__get_errors_from_substatus" and update the function description accordingly

Reply:
done

Reply:
@rane-rajasi to resolve if satisfied.